azurerm/resource_arm_subnet.go

```diff
@@ -204,6 +245,11 @@ func resourceArmSubnetRead(d *schema.ResourceData, meta interface{}) error {
 		if err := d.Set("service_endpoints", serviceEndpoints); err != nil {
 			return err
 		}
+
+		delegation := flattenSubnetDelegation(props.Delegations)
+		if err := d.Set("delegation", delegation); err != nil {
+			return err
```

Comment:
can we wrap this to explain what failed (this is more for our benefit during bug reports than anything else, so the context is helpful)

```suggestion
			return fmt.Errorf("Error flattening `delegation`: %+v", err)
```